File: code-examples/quick-sort/quick-sort.php
class Solution {
    function sortArray($nums) {
        if(count($nums) <= 1) {
            return $nums;
        }

        $low = $high = [];
        $pivot = $nums[0];
        for ($i = 1; $i < count($nums); $i++) {
            if($nums[$i] > $pivot) {
                $high[] = $nums[$i];
            } elseif($nums[$i] < $pivot) {
                $low[] = $nums[$i];
            }
        }
        
        return [...$this->sortArray($low), $pivot, ...$this->sortArray($high)];
    }
}

File: code-examples/quick-sort/quick-sort.java
import java.util.ArrayList;

class Solution {
    public ArrayList<Integer> sortArray(ArrayList<Integer> nums) {
        if(nums.size() <= 1) {
            return nums;
        }

        ArrayList<Integer> low = new ArrayList<>();
        ArrayList<Integer> high = new ArrayList<>();
        int pivot = nums.getFirst();

        for(int i = 1; i < nums.size(); i++) {
            if(nums.get(i) < pivot) {
                low.add(nums.get(i));
            } else if(nums.get(i) > pivot){
                high.add(nums.get(i));
            }
        }

        ArrayList<Integer>result = new ArrayList<>(this.sortArray(low));
        result.add(pivot);
        result.addAll(this.sortArray(high));

        return result;
    }
}
